<!doctype html><html lang=en class=no-js> <head><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><meta name=description content="tRNA abundance quantification for Colin Wu Lab using mim-tRNASeq toolkit"><meta name=author content="Vishal Koparde, Ph.D."><link rel=icon href=../../assets/images/favicon.png><meta name=generator content="mkdocs-1.3.0, mkdocs-material-8.2.16"><title>Background - tRNA abundance quantification for Colin Wu Lab (RBL)</title><link rel=stylesheet href=../../assets/stylesheets/main.1c3799f8.min.css><link rel=stylesheet href=../../assets/stylesheets/palette.cc9b2e1e.min.css><link rel=preconnect href=https://fonts.gstatic.com crossorigin><link rel=stylesheet href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback"><style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style><script>__md_scope=new URL("../..",location),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script></head> <body dir=ltr data-md-color-scheme=default data-md-color-primary=indigo data-md-color-accent=indigo> <script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script> <input class=md-toggle data-md-toggle=drawer type=checkbox id=__drawer autocomplete=off> <input class=md-toggle data-md-toggle=search type=checkbox id=__search autocomplete=off> <label class=md-overlay for=__drawer></label> <div data-md-component=skip> <a href=#background class=md-skip> Skip to content </a> </div> <div data-md-component=announce> </div> <div data-md-component=outdated hidden> <aside class="md-banner md-banner--warning"> </aside> </div> <header class=md-header data-md-component=header> <nav class="md-header__inner md-grid" aria-label=Header> <a href=../.. title="tRNA abundance quantification for Colin Wu Lab (RBL)" class="md-header__button md-logo" aria-label="tRNA abundance quantification for Colin Wu Lab (RBL)" data-md-component=logo> <img src=../../assets/images/doc-book.svg alt=logo> </a> <label class="md-header__button md-icon" for=__drawer> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg> </label> <div class=md-header__title data-md-component=header-title> <div class=md-header__ellipsis> <div class=md-header__topic> <span class=md-ellipsis> tRNA abundance quantification for Colin Wu Lab (RBL) </span> </div> <div class=md-header__topic data-md-component=header-topic> <span class=md-ellipsis> Background </span> </div> </div> </div> <form class=md-header__option data-md-component=palette> <input class=md-option data-md-color-media data-md-color-scheme=default data-md-color-primary=indigo data-md-color-accent=indigo aria-label="Switch to dark mode" type=radio name=__palette id=__palette_1> <label class="md-header__button md-icon" title="Switch to dark mode" for=__palette_2 hidden> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M17 6H7c-3.31 0-6 2.69-6 6s2.69 6 6 6h10c3.31 0 6-2.69 6-6s-2.69-6-6-6zm0 10H7c-2.21 0-4-1.79-4-4s1.79-4 4-4h10c2.21 0 4 1.79 4 4s-1.79 4-4 4zM7 9c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"/></svg> </label> <input class=md-option data-md-color-media data-md-color-scheme=slate data-md-color-primary=red data-md-color-accent=red aria-label="Switch to light mode" type=radio name=__palette id=__palette_2> <label class="md-header__button md-icon" title="Switch to light mode" for=__palette_1 hidden> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M17 7H7a5 5 0 0 0-5 5 5 5 0 0 0 5 5h10a5 5 0 0 0 5-5 5 5 0 0 0-5-5m0 8a3 3 0 0 1-3-3 3 3 0 0 1 3-3 3 3 0 0 1 3 3 3 3 0 0 1-3 3Z"/></svg> </label> </form> <label class="md-header__button md-icon" for=__search> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg> </label> <div class=md-search data-md-component=search role=dialog> <label class=md-search__overlay for=__search></label> <div class=md-search__inner role=search> <form class=md-search__form name=search> <input type=text class=md-search__input name=query aria-label=Search placeholder=Search autocapitalize=off autocorrect=off autocomplete=off spellcheck=false data-md-component=search-query required> <label class="md-search__icon md-icon" for=__search> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg> </label> <nav class=md-search__options aria-label=Search> <button type=reset class="md-search__icon md-icon" aria-label=Clear tabindex=-1> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg> </button> </nav> </form> <div class=md-search__output> <div class=md-search__scrollwrap data-md-scrollfix> <div class=md-search-result data-md-component=search-result> <div class=md-search-result__meta> Initializing search </div> <ol class=md-search-result__list></ol> </div> </div> </div> </div> </div> <div class=md-header__source> <a href=https://github.com/RBL-NCI/Wu_tRNA_seq title="Go to repository" class=md-source data-md-component=source> <div class="md-source__icon md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 448 512"><!-- Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg> </div> <div class=md-source__repository> CCBR/HowTos </div> </a> </div> </nav> </header> <div class=md-container data-md-component=container> <nav class=md-tabs aria-label=Tabs data-md-component=tabs> <div class="md-tabs__inner md-grid"> <ul class=md-tabs__list> <li class=md-tabs__item> <a href=../.. class=md-tabs__link> Intro </a> </li> <li class=md-tabs__item> <a href=./ class="md-tabs__link md-tabs__link--active"> Background </a> </li> <li class=md-tabs__item> <a href=../datasets/ class=md-tabs__link> Datasets </a> </li> </ul> </div> </nav> <main class=md-main data-md-component=main> <div class="md-main__inner md-grid"> <div class="md-sidebar md-sidebar--primary" data-md-component=sidebar data-md-type=navigation> <div class=md-sidebar__scrollwrap> <div class=md-sidebar__inner> <nav class="md-nav md-nav--primary md-nav--lifted md-nav--integrated" aria-label=Navigation data-md-level=0> <label class=md-nav__title for=__drawer> <a href=../.. title="tRNA abundance quantification for Colin Wu Lab (RBL)" class="md-nav__button md-logo" aria-label="tRNA abundance quantification for Colin Wu Lab (RBL)" data-md-component=logo> <img src=../../assets/images/doc-book.svg alt=logo> </a> tRNA abundance quantification for Colin Wu Lab (RBL) </label> <div class=md-nav__source> <a href=https://github.com/RBL-NCI/Wu_tRNA_seq title="Go to repository" class=md-source data-md-component=source> <div class="md-source__icon md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 448 512"><!-- Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg> </div> <div class=md-source__repository> CCBR/HowTos </div> </a> </div> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../.. class=md-nav__link> Intro </a> </li> <li class="md-nav__item md-nav__item--active"> <input class="md-nav__toggle md-toggle" data-md-toggle=toc type=checkbox id=__toc> <label class="md-nav__link md-nav__link--active" for=__toc> Background <span class="md-nav__icon md-icon"></span> </label> <a href=./ class="md-nav__link md-nav__link--active"> Background </a> <nav class="md-nav md-nav--secondary" aria-label="Table of contents"> <label class=md-nav__title for=__toc> <span class="md-nav__icon md-icon"></span> Table of contents </label> <ul class=md-nav__list data-md-component=toc data-md-scrollfix> <li class=md-nav__item> <a href=#background class=md-nav__link> Background </a> </li> <li class=md-nav__item> <a href=#mimseq-and-its-issues class=md-nav__link> mimseq and its issues </a> <nav class=md-nav aria-label="mimseq and its issues"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#group-argument-must-be-none-for-now class=md-nav__link> "group argument must be None for now" </a> </li> <li class=md-nav__item> <a href=#keyerror class=md-nav__link> Keyerror </a> </li> </ul> </nav> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=../datasets/ class=md-nav__link> Datasets </a> </li> </ul> </nav> </div> </div> </div> <div class=md-content data-md-component=content> <article class="md-content__inner md-typeset"> <a href=https://github.com/RBL-NCI/Wu_tRNA_seq/edit/docs/docs/wu_rbl/background.md title="Edit this page" class="md-content__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25Z"/></svg> </a> <h1>Background</h1> <h3 id=background>Background<a class=headerlink href=#background title="Permanent link">&para;</a></h3> <p>tRNA quantification is hard. cDNA synthesis is hampered by: * pervasive RT blocks by modified (mostly methylated) nucleosides ... this leads to premature RT stops at modified sites... resulting in very short reads. * extensive similarity between tRNA genes ... eg. tRNA genes coding for 2 different codons can have only a single nucleotide difference. ... this make the <em>alignment</em> step of analysis pipeline extremely difficult.</p> <p>The composition of cellular tRNA pools is critical for efficient mRNA decoding and proteome integrity. Defective tRNA biogenesis is linked to neurological disorders and cancer.</p> <p><u>AlkB</u>: AlkB is a protein found in E. coli which removes alkylation groups from ssDNA.</p> <p><u>Isodecoder</u>: tRNA genes that share the same anticodon but have different body sequences have been termed as “isodecoders”, implying that they all read the same codon in translation.</p> <p><u>tRNA human genes</u>: Facts:</p> <ul> <li>Number of different tRNA anticodon families: 57</li> <li>Number of unique tRNA molecules produced (all cell types included): ~400</li> </ul> <p><u>K562</u>: Human CML (chronic myelogenous leukemia) cell line</p> <h3 id=mimseq-and-its-issues>mimseq and its issues<a class=headerlink href=#mimseq-and-its-issues title="Permanent link">&para;</a></h3> <p><a href=https://www.sciencedirect.com/science/article/pii/S1097276521000484>mimseq</a> is the analysis tool to be used for quantifying tRNA abundance for Wu lab samples (K562 cell line samples). Code can be found <a href=https://github.com/nedialkova-lab/mim-tRNAseq>here</a></p> <h4 id=group-argument-must-be-none-for-now>"group argument must be None for now"<a class=headerlink href=#group-argument-must-be-none-for-now title="Permanent link">&para;</a></h4> <p>After installation, got this error:</p> <div class=highlight><pre><span></span><code>% mimseq  --species Hsap  --cluster-id <span class=m>0</span>.95  --threads <span class=m>15</span>  --min-cov <span class=m>0</span>.0005  --max-mismatches <span class=m>0</span>.
<span class=m>1</span>  --control-condition HEK293T  -n hg38_test  --out-dir hg38_HEK239vsK562  --max-multi <span class=m>4</span> --rem
ap  --remap-mismatches <span class=m>0</span>.075  sampleData_HEKvsK562.txt
           _                 _   ____  _   _    _
 _ __ ___ <span class=o>(</span>_<span class=o>)</span>_ __ ___       <span class=p>|</span> <span class=p>|</span>_<span class=p>|</span>  _ <span class=se>\|</span> <span class=se>\ </span><span class=p>|</span> <span class=p>|</span>  / <span class=se>\ </span>  ___  ___  __ _
<span class=p>|</span> <span class=s1>&#39;_ ` _ \| | &#39;</span>_ <span class=sb>`</span> _ <span class=se>\ </span>_____<span class=p>|</span> __<span class=p>|</span> <span class=p>|</span>_<span class=o>)</span> <span class=p>|</span>  <span class=se>\|</span> <span class=p>|</span> / _ <span class=se>\ </span>/ __<span class=p>|</span>/ _ <span class=se>\/</span> _<span class=sb>`</span> <span class=p>|</span>
<span class=p>|</span> <span class=p>|</span> <span class=p>|</span> <span class=p>|</span> <span class=p>|</span> <span class=p>|</span> <span class=p>|</span> <span class=p>|</span> <span class=p>|</span> <span class=p>|</span> <span class=p>|</span> <span class=p>|</span>_____<span class=p>|</span> <span class=p>|</span>_<span class=p>|</span>  _ &lt;<span class=p>|</span> <span class=p>|</span><span class=se>\ </span> <span class=p>|</span>/ ___ <span class=se>\\</span>__ <span class=se>\ </span> __/ <span class=o>(</span>_<span class=p>|</span> <span class=p>|</span>
<span class=p>|</span>_<span class=p>|</span> <span class=p>|</span>_<span class=p>|</span> <span class=p>|</span>_<span class=p>|</span>_<span class=p>|</span>_<span class=p>|</span> <span class=p>|</span>_<span class=p>|</span> <span class=p>|</span>_<span class=p>|</span>      <span class=se>\_</span>_<span class=p>|</span>_<span class=p>|</span> <span class=se>\_\_</span><span class=p>|</span> <span class=se>\_</span>/_/   <span class=se>\_\_</span>__/<span class=se>\_</span>__<span class=p>|</span><span class=se>\_</span>_, <span class=p>|</span>
                                                                 <span class=p>|</span>_<span class=p>|</span>

 Modification-induced misincorporation analysis of tRNA sequencing data

<span class=m>2022</span>-06-27 <span class=m>15</span>:51:26,542 <span class=o>[</span>INFO <span class=o>]</span> mim-tRNAseq v1.1.6 run with command:
<span class=m>2022</span>-06-27 <span class=m>15</span>:51:26,542 <span class=o>[</span>INFO <span class=o>]</span> /usr/local/bin/mimseq --species Hsap --cluster-id <span class=m>0</span>.95 --threads <span class=m>15</span> --min-cov <span class=m>0</span>.0005 --max-mismatches <span class=m>0</span>.1 --control-condition HEK293T -n hg38_test --out-dir hg38_HEK239vsK562 --max-multi <span class=m>4</span> --remap --remap-mismatches <span class=m>0</span>.075 sampleData_HEKvsK562.txt
<span class=m>2022</span>-06-27 <span class=m>15</span>:51:26,544 <span class=o>[</span>INFO <span class=o>]</span>
...
...
+--------------------------------------------------------------------+
<span class=p>|</span> Analysing misincorporations and stops to RT, and analysing <span class=m>3</span><span class=s1>&#39; ends |</span>
<span class=s1>+--------------------------------------------------------------------+</span>
<span class=s1>2022-06-27 15:56:15,178 [INFO ] ** Discovering unannotated modifications for realignment **</span>
<span class=s1>Traceback (most recent call last):</span>
<span class=s1>  File &quot;/usr/local/bin/mimseq&quot;, line 8, in &lt;module&gt;</span>
<span class=s1>    sys.exit(main())</span>
<span class=s1>  File &quot;/usr/local/lib/python3.8/dist-packages/mimseq/mimseq.py&quot;, line 407, in main</span>
<span class=s1>    mimseq(args.trnas, args.trnaout, args.name, args.species, args.out, args.cluster, args.cluster_id, args.cov_diff, \</span>
<span class=s1>  File &quot;/usr/local/lib/python3.8/dist-packages/mimseq/mimseq.py&quot;, line 132, in mimseq</span>
<span class=s1>    new_mods, new_Inosines, filtered_cov, filter_warning, unsplitCluster_lookup,readRef_unsplit_newNames = generateModsTable(coverageData, out, name, threads, min_cov, mismatch_dict, insert_dict, del_dict, cluster_dict, cca, remap, misinc_thresh, mod_lists, Inosine_lists, tRNA_dict, Inosine_clusters, unique_isodecoderMMs_new, splitBool_new, isodecoder_sizes, unsplitCluster_lookup, cluster)</span>
<span class=s1>  File &quot;/usr/local/lib/python3.8/dist-packages/mimseq/mmQuant.py&quot;, line 634, in generateModsTable</span>
<span class=s1>    pool = MyPool(multi)</span>
<span class=s1>  File &quot;/usr/lib/python3.8/multiprocessing/pool.py&quot;, line 212, in __init__</span>
<span class=s1>    self._repopulate_pool()</span>
<span class=s1>  File &quot;/usr/lib/python3.8/multiprocessing/pool.py&quot;, line 303, in _repopulate_pool</span>
<span class=s1>    return self._repopulate_pool_static(self._ctx, self.Process,</span>
<span class=s1>  File &quot;/usr/lib/python3.8/multiprocessing/pool.py&quot;, line 319, in _repopulate_pool_static</span>
<span class=s1>    w = Process(ctx, target=worker,</span>
<span class=s1>  File &quot;/usr/lib/python3.8/multiprocessing/process.py&quot;, line 82, in __init__</span>
<span class=s1>    assert group is None, &#39;</span>group argument must be None <span class=k>for</span> now<span class=err>&#39;</span>
AssertionError: group argument must be None <span class=k>for</span> now
</code></pre></div> <p>This was resolved with specific version of python/conda/mamba. See <a href=https://github.com/nedialkova-lab/mim-tRNAseq/issues/33>here</a> for details.</p> <h4 id=keyerror>Keyerror<a class=headerlink href=#keyerror title="Permanent link">&para;</a></h4> <p>With the above issue fixed, another error was found</p> <div class=highlight><pre><span></span><code>+-----------------------------------+
<span class=p>|</span> Final deconvolution and filtering <span class=p>|</span>
+-----------------------------------+
<span class=m>2022</span>-07-10 <span class=m>12</span>:04:32,132 <span class=o>[</span>INFO <span class=o>]</span> <span class=m>234</span> clusters filtered out according to minimum coverage threshold: <span class=m>0</span>.05% of total tRNA coverage.
<span class=m>2022</span>-07-10 <span class=m>12</span>:05:19,876 <span class=o>[</span>INFO <span class=o>]</span> Output final tables, counts and new mods...
<span class=m>2022</span>-07-10 <span class=m>12</span>:05:23,815 <span class=o>[</span>INFO <span class=o>]</span> ** Read counts per anticodon saved to Tet2KOvsWT/counts/Anticodon_counts_raw.txt **
<span class=m>2022</span>-07-10 <span class=m>12</span>:05:23,815 <span class=o>[</span>INFO <span class=o>]</span> ** Read counts per isodecoder saved to Tet2KOvsWT/counts/Isodecoder_counts_raw.txt **
<span class=m>2022</span>-07-10 <span class=m>12</span>:05:23,815 <span class=o>[</span>INFO <span class=o>]</span> <span class=m>11</span> sequences could not be deconvoluted as &gt;10% parent assigned-reads <span class=k>do</span> not match parent sequence!
<span class=m>2022</span>-07-10 <span class=m>12</span>:05:23,815 <span class=o>[</span>INFO <span class=o>]</span> Reasons <span class=k>for</span> this include misaligment of reads to an incorrect cluster, or inaccurate aligment by GSNAP <span class=o>(</span>e.g. at indels <span class=k>in</span> reads<span class=o>)</span> prohibiting correct devonvolution.
<span class=m>2022</span>-07-10 <span class=m>12</span>:05:23,815 <span class=o>[</span>INFO <span class=o>]</span> <span class=m>79</span> total unique sequences not deconvoluted due to mismatches at modified sites, insufficient coverage or <span class=nb>read</span> mismatches to parent
Traceback <span class=o>(</span>most recent call last<span class=o>)</span>:
  File <span class=s2>&quot;/opt2/conda/envs/mimseq/bin/mimseq&quot;</span>, line <span class=m>10</span>, <span class=k>in</span> &lt;module&gt;
    sys.exit<span class=o>(</span>main<span class=o>())</span>
  File <span class=s2>&quot;/opt2/conda/envs/mimseq/lib/python3.7/site-packages/mimseq/mimseq.py&quot;</span>, line <span class=m>410</span>, <span class=k>in</span> main
    args.misinc_thresh, args.mito, args.pretrnas, args.local_mod, args.p_adj, args.sampledata<span class=o>)</span>
  File <span class=s2>&quot;/opt2/conda/envs/mimseq/lib/python3.7/site-packages/mimseq/mimseq.py&quot;</span>, line <span class=m>161</span>, <span class=k>in</span> mimseq
    <span class=nv>isodecoder_sizes</span> <span class=o>=</span> writeIsodecoderInfo<span class=o>(</span>out, name, isodecoder_sizes,readRef_unsplit_newNames, tRNA_dict<span class=o>)</span>
  File <span class=s2>&quot;/opt2/conda/envs/mimseq/lib/python3.7/site-packages/mimseq/splitClusters.py&quot;</span>, line <span class=m>477</span>, <span class=k>in</span> writeIsodecoderInfo
    del isodecoder_sizes<span class=o>[</span>gene<span class=o>]</span>
KeyError: <span class=s1>&#39;Homo_sapiens_tRNA-Glu-TTC-1-1&#39;</span>
</code></pre></div> <p>This issue was reported by other users as well and the author recommended using the latest updated version of <code>splitClusters.py</code>. See <a href=https://github.com/nedialkova-lab/mim-tRNAseq/issues/34>here</a> for details.</p> <hr> <div class=md-source-file> <small> Last update: 2022-07-29 </small> </div> </article> </div> </div> <a href=# class="md-top md-icon" data-md-component=top data-md-state=hidden> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg> Back to top </a> </main> <footer class=md-footer> <nav class="md-footer__inner md-grid" aria-label=Footer> <a href=../.. class="md-footer__link md-footer__link--prev" aria-label="Previous: Intro" rel=prev> <div class="md-footer__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg> </div> <div class=md-footer__title> <div class=md-ellipsis> <span class=md-footer__direction> Previous </span> Intro </div> </div> </a> <a href=../datasets/ class="md-footer__link md-footer__link--next" aria-label="Next: Datasets" rel=next> <div class=md-footer__title> <div class=md-ellipsis> <span class=md-footer__direction> Next </span> Datasets </div> </div> <div class="md-footer__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4Z"/></svg> </div> </a> </nav> <div class="md-footer-meta md-typeset"> <div class="md-footer-meta__inner md-grid"> <div class=md-copyright> <div class=md-copyright__highlight> Copyright &copy; 2022 CCBR </div> </div> <div class=md-social> <a href=https://ccbr.ccr.cancer.gov/ target=_blank rel=noopener title=ccbr.ccr.cancer.gov class=md-social__link> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 640 512"><!-- Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M319.9 320c57.41 0 103.1-46.56 103.1-104s-46.54-104-103.1-104c-57.41 0-103.1 46.56-103.1 104-.9 57.4 45.7 104 103.1 104zm50 32h-99.8C191.6 352 128 411.7 128 485.3c0 14.8 12.7 26.7 28.4 26.7h327.2c15.7 0 28.4-11.9 28.4-26.7 0-73.6-63.6-133.3-142.1-133.3zM512 160c44.18 0 80-35.82 80-80S556.2 0 512 0c-44.18 0-80 35.82-80 80s35.8 80 80 80zm-328.1 56c0-5.449.982-10.63 1.609-15.91C174.6 194.1 162.6 192 149.9 192H88.08C39.44 192 0 233.8 0 285.3 0 295.6 7.887 304 17.62 304h199.5c-20.42-23.8-33.22-54.3-33.22-88zM128 160c44.18 0 80-35.82 80-80S172.2 0 128 0C83.82 0 48 35.82 48 80s35.82 80 80 80zm423.9 32h-61.84c-12.8 0-24.88 3.037-35.86 8.24.6 5.26 1.6 10.36 1.6 15.76 0 33.71-12.78 64.21-33.16 88h199.7c9.76 0 17.66-8.4 17.66-18.7 0-51.5-39.4-93.3-88.1-93.3z"/></svg> </a> <a href=https://github.com/CCBR target=_blank rel=noopener title=github.com class=md-social__link> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 496 512"><!-- Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg> </a> <a href=mailto:vishal.koparde@nih.gov target=_blank rel=noopener title class=md-social__link> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 512 512"><!-- Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="m511.6 36.86-64 415.1a32.008 32.008 0 0 1-31.65 27.147c-4.188 0-8.319-.815-12.29-2.472l-122.6-51.1-50.86 76.29C226.3 508.5 219.8 512 212.8 512c-11.5 0-20.8-9.3-20.8-20.8v-96.18c0-7.115 2.372-14.03 6.742-19.64L416 96 122.3 360.3 19.69 317.5C8.438 312.8.812 302.2.062 289.1s5.47-23.72 16.06-29.77l448-255.1c10.69-6.109 23.88-5.547 34 1.406S513.5 24.72 511.6 36.86z"/></svg> </a> </div> </div> </div> </footer> </div> <div class=md-dialog data-md-component=dialog> <div class="md-dialog__inner md-typeset"></div> </div> <script id=__config type=application/json>{"base": "../..", "features": ["navigation.tabs", "navigation.top", "navigation.indexes", "toc.integrate"], "search": "../../assets/javascripts/workers/search.2a1c317c.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version.title": "Select version"}, "version": {"provider": "mike"}}</script> <script src=../../assets/javascripts/bundle.3a4b43e5.min.js></script> </body> </html>